MediaBrowser.WebDashboard/dashboard-ui/bower_components/emby-webcomponents/browserdeviceprofile.js

```diff
@@ -1,384 +1,916 @@
-            videoTestElement = document.createElement("video"),
-            canPlayVp8 = videoTestElement.canPlayType('video/webm; codecs="vp8"').replace(/no/, ""),
-            canPlayVp9 = videoTestElement.canPlayType('video/webm; codecs="vp9"').replace(/no/, ""),
-            webmAudioCodecs = ["vorbis"],
-            canPlayMkv = testCanPlayMkv(videoTestElement),
-            profile = {};
-        profile.MaxStreamingBitrate = 12e7, profile.MaxStaticBitrate = 1e8, profile.MusicStreamingTranscodingBitrate = Math.min(12e7, 192e3), profile.DirectPlayProfiles = [];
-        var videoAudioCodecs = [],
-            hlsVideoAudioCodecs = [],
-            supportsMp3VideoAudio = videoTestElement.canPlayType('video/mp4; codecs="avc1.640029, mp4a.69"').replace(/no/, "") || videoTestElement.canPlayType('video/mp4; codecs="avc1.640029, mp4a.6B"').replace(/no/, ""),
-            supportsMp2VideoAudio = browser.edgeUwp || browser.tizen || browser.orsay || browser.web0s,
-            maxVideoWidth = browser.xboxOne && self.screen ? self.screen.width : null;
-        options.maxVideoWidth && (maxVideoWidth = options.maxVideoWidth);
-        var canPlayAacVideoAudio = videoTestElement.canPlayType('video/mp4; codecs="avc1.640029, mp4a.40.2"').replace(/no/, "");
-        if (canPlayAacVideoAudio && browser.chromecast && physicalAudioChannels <= 2 && videoAudioCodecs.push("aac"), supportsAc3(videoTestElement)) {
-            videoAudioCodecs.push("ac3");
+        var physicalAudioChannels = options.audioChannels || (browser.tv || browser.ps4 || browser.xboxOne ? 6 : 2);
+
+        var bitrateSetting = getMaxBitrate();
+
+        var videoTestElement = document.createElement('video');
+
+        var canPlayVp8 = videoTestElement.canPlayType('video/webm; codecs="vp8"').replace(/no/, '');
+        var canPlayVp9 = videoTestElement.canPlayType('video/webm; codecs="vp9"').replace(/no/, '');
+        var webmAudioCodecs = ['vorbis'];
+
+        var canPlayMkv = testCanPlayMkv(videoTestElement);
+
+        var profile = {};
+
+        profile.MaxStreamingBitrate = bitrateSetting;
+        profile.MaxStaticBitrate = 100000000;
+        profile.MusicStreamingTranscodingBitrate = Math.min(bitrateSetting, 192000);
+
+        profile.DirectPlayProfiles = [];
+
+        var videoAudioCodecs = [];
+        var hlsVideoAudioCodecs = [];
+
+        var supportsMp3VideoAudio = videoTestElement.canPlayType('video/mp4; codecs="avc1.640029, mp4a.69"').replace(/no/, '') ||
+            videoTestElement.canPlayType('video/mp4; codecs="avc1.640029, mp4a.6B"').replace(/no/, '');
+
+        // Not sure how to test for this
+        var supportsMp2VideoAudio = browser.edgeUwp || browser.tizen || browser.orsay || browser.web0s;
+
+        var maxVideoWidth = browser.xboxOne ?
+            (self.screen ? self.screen.width : null) :
+            null;
+
+        if (options.maxVideoWidth) {
+            maxVideoWidth = options.maxVideoWidth;
```

Comment:
shouldn't we check if `options.maxVideoWidth` is less or equal to `maxVideoWidth` if latter is not `null`?..